internal/databases/mongo/binary/checksum.go

```diff
@@ -0,0 +1,32 @@
+package binary
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"hash"
+)
+
+type ChecksumCalculator struct {
```

Comment:
Can we extract this directly to the `internal` package? It looks like a not mongo-specific code.